src/interfaces.js

```diff
@@ -73,5 +73,65 @@ interface RouteConfig {
   */
   caseSensitive?: boolean;
 
+  /**
+  * Add to specify an activation strategy if it is always the same and you do not want that
+  * to be in your view-model code. Available values are 'replace' and 'invoke-lifecycle'.
+  */
+  activationStrategy?: string;
+  
   [x: string]: any;
 }
+
+type ShouldContinue = boolean|Promise<boolean>|PromiseLike<boolean>|NavigationCommand;
+type Potential = Promise<void>|PromiseLike<void>|IObservable;
```

Comment:
I guess it makes sense to add type info (`Potential`) to `processPotential` function as well - [activation.js#L222](https://github.com/aurelia/router/blob/e7d01b7fd73adf90feb2dc2301cdf372a8209b48/src/activation.js#L222)